x-pack/plugins/security_solution/public/resolver/view/submenu.tsx

```diff
@@ -52,73 +43,46 @@ interface ResolverSubmenuOption {
 
 export type ResolverSubmenuOptionList = ResolverSubmenuOption[] | string;
 
-const OptionListItem = styled.div`
-  width: 175px;
-`;
-
-const OptionList = React.memo(
+/**
+ * This will be the "host button" that displays the "total number of related events" and opens
+ * the sumbmenu (with counts by category) when clicked.
+ */
+const DetailHostButton = React.memo(
   ({
-    subMenuOptions,
-    isLoading,
+    hasMenu,
+    menuIsOpen,
+    action,
+    count,
+    title,
+    buttonBorderColor,
+    nodeID,
   }: {
-    subMenuOptions: ResolverSubmenuOptionList;
-    isLoading: boolean;
+    hasMenu: boolean;
+    menuIsOpen?: boolean;
+    action: (evt: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
+    count?: number;
+    title: string;
+    buttonBorderColor: ButtonColor;
+    nodeID: string;
   }) => {
-    const [options, setOptions] = useState<EuiSelectableOption[]>(() =>
-      typeof subMenuOptions !== 'object'
-        ? []
-        : subMenuOptions.map((option: ResolverSubmenuOption) => {
-            const dataTestSubj = 'resolver:map:node-submenu-item';
-            return option.prefix
-              ? {
-                  label: option.optionTitle,
-                  prepend: <span>{option.prefix} </span>,
-                  'data-test-subj': dataTestSubj,
-                }
-              : {
-                  label: option.optionTitle,
-                  prepend: <span />,
-                  'data-test-subj': dataTestSubj,
-                };
-          })
-    );
-
-    const actionsByLabel: Record<string, () => unknown> = useMemo(() => {
-      if (typeof subMenuOptions !== 'object') {
-        return {};
-      }
-      return subMenuOptions.reduce((titleActionRecord, opt) => {
-        const { optionTitle, action } = opt;
-        return { ...titleActionRecord, [optionTitle]: action };
-      }, {});
-    }, [subMenuOptions]);
-
-    const selectableProps = useMemo(() => {
-      return {
-        listProps: { showIcons: true, bordered: true },
-        onChange: (newOptions: EuiSelectableOption[]) => {
-          const selectedOption = newOptions.find((opt) => opt.checked === 'on');
-          if (selectedOption) {
-            const { label } = selectedOption;
-            const actionToTake = actionsByLabel[label];
-            if (typeof actionToTake === 'function') {
-              actionToTake();
-            }
-          }
-          setOptions(newOptions);
-        },
-      };
-    }, [actionsByLabel]);
-
+    const hasIcon = hasMenu ?? false;
+    const iconType = menuIsOpen === true ? 'arrowUp' : 'arrowDown';
     return (
-      <EuiSelectable
-        singleSelection={true}
-        options={options}
-        {...selectableProps}
-        isLoading={isLoading}
+      <EuiButton
+        onClick={action}
+        iconType={hasIcon ? iconType : 'none'}
+        fill={false}
+        color={'primary'}
+        style={{ height: 'fit-content', lineHeight: 1, padding: '.25em', fontSize: '.85rem' }}
```

Comment:
For multiple values like this, could we move it into a styled component?

Reply:
https://github.com/elastic/kibana/pull/76997/commits/18598d28d605b666e981e91ad8962ac5f6b640e2 Done here, thanks!